Sources/Tracking/CustomerIOImplementation+ScreenViews.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+#if canImport(UIKit)
+import UIKit
+
+public extension CustomerIOImplementation {
+    func enableAutoScreenviewTracking() {
+        let selector1 = #selector(UIViewController.viewDidAppear(_:))
+        let selector2 = #selector(CustomerIOImplementation._swizzled_UIKit_viewDidAppear(_:))
+        guard let originalMethod = class_getInstanceMethod(UIViewController.self, selector1) else {
+            return
+        }
+        guard let swizzleMethod = class_getInstanceMethod(CustomerIOImplementation.self, selector2) else {
+            return
+        }
+        method_exchangeImplementations(originalMethod, swizzleMethod)
+    }
+
+    @objc dynamic func _swizzled_UIKit_viewDidAppear(_ animated: Bool) {
+        _swizzled_UIKit_viewDidAppear(animated)
+
+        guard let delegate = UIApplication.shared.delegate else {
+            return
+        }
+
+        guard let window = delegate.window else {
+            return
+        }
+
+        var viewController = window!.rootViewController
+        if viewController is UINavigationController {
+            viewController = (viewController as! UINavigationController).visibleViewController
+        }
+
+        var name = String(describing: type(of: viewController)).replacingOccurrences(of: "ViewController", with: "")
+
+        if name.isEmpty || name == "" {
+            if viewController?.title != nil {
+                name = viewController?.title ?? "Unknown"
+            }
+            if name.isEmpty || name == "" {
+                // Could not infer screen name
+                name = "Unknown"
+            }
+        }
```

Comment:
Can be reduced to

`if (name.isEmpty || name == "") { 
if let vcTitle = viewController?.title {
name = vcTitle
}else {
name = "unknown"
}
}`

Sorry for bad indentation here. 😅

Reply:
thanks! switched to using a slightly more swift-y approach that tidies it